src/server/search/doc_accessors.cc

```diff
@@ -107,6 +137,9 @@ SearchDocData JsonAccessor::Serialize(search::Schema schema) const {
   return out;
 }
 
+thread_local absl::flat_hash_map<std::string, std::unique_ptr<JsonAccessor::JsonPathContainer>>
```

Comment:
Why do we need `unique_ptr` for the `value_type` here?

Reply:
I don't want to pull in json dependencies into the header, so I forward declared JsonPathContainer. Actually pulling in json deps is not crucial as this header is used only in search internal files 🤔

Reply:
Having global caches with no eviction and no limit seems dangerous, memory wise.
Could we turn this into an LRU? Or bind this caching to the lifetime of the query?

Reply:
Those are field names defined by developers. Each schema has usually a few fields, and there are usually no more than a few indices per database. Having them occupy kilobytes of memory would require developers to type out kilobytes of field names which is really unlikely

We can clear this cache once indices are deleted

Reply:
Ah, they can't query for arbitrary fields outside of established indices? Somehow I thought it's possible but just slower.

Reply:
No, its only used for accessing an index - and that index must have been explicitly described by the user